homeassistant/components/notion/binary_sensor.py

```diff
@@ -48,27 +93,20 @@ async def async_setup_entry(
     """Set up Notion sensors based on a config entry."""
     coordinator = hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id]
 
-    sensor_list = []
-    for task_id, task in coordinator.data["tasks"].items():
-        if task["task_type"] not in BINARY_SENSOR_TYPES:
-            continue
-
-        name, device_class = BINARY_SENSOR_TYPES[task["task_type"]]
-        sensor = coordinator.data["sensors"][task["sensor_id"]]
-
-        sensor_list.append(
+    async_add_entities(
+        [
             NotionBinarySensor(
                 coordinator,
                 task_id,
-                sensor["id"],
-                sensor["bridge"]["id"],
-                sensor["system_id"],
-                name,
-                device_class,
+                coordinator.data["sensors"][task["sensor_id"]]["id"],
+                coordinator.data["sensors"][task["sensor_id"]]["bridge"]["id"],
+                coordinator.data["sensors"][task["sensor_id"]]["system_id"],
```

Comment:
You should be able to use an assignment expression to make this a bit better. I believe this here would work:
```py
        [
            NotionBinarySensor(
                coordinator,
                task_id,
                sensor["id"],
                sensor["bridge"]["id"],
                sensor["system_id"],
                BINARY_SENSOR_DESCRIPTIONS[task["task_type"]],
            )
            for task_id, task in coordinator.data["tasks"].items()
            if task["task_type"] in BINARY_SENSOR_DESCRIPTIONS
            and (sensor := coordinator.data["sensors"][task["sensor_id"]])
        ]
```